grails-app/services/org/pih/warehouse/inventory/CycleCountService.groovy

```diff
@@ -484,13 +483,8 @@ class CycleCountService {
      * when we want to trigger a status recalculation outside of that flow, such as when the status of a cycle
      * count item changes.
      */
-    private void recomputeCycleCountStatus(CycleCount cycleCount) {
+    private static void recomputeCycleCountStatus(CycleCount cycleCount) {
         cycleCount.status = cycleCount.recomputeStatus()
-
-        // TODO: We sometimes get errors and the status update isn't persisted unless we call save like this. GORM
-        //       should be automatically persisting the status update when its session is flushed so we shouldn't
-        //       need this save() but something weird is happening. Investigate why this line is needed.
-        cycleCount.save()
```

Comment:
After this change, it feels like this `recomputeCycleCountStatus` is an unnecessary method. Why we can't do 
```
cycleCount.status = cycleCount.recomputeStatus()
```
directly in the line that calls this one?

Reply:
probably due to my laziness :smile: it's used in a few methods, so I didn't want to refactor the other methods and make sure they still work (just in case). I can remove it if you strongly feel it's "form over substance"